lib/runtime/wasmer/exports_test.go

```diff
@@ -1087,3 +1087,126 @@ func newTrieFromPairs(t *testing.T, filename string) *trie.Trie {
 	require.NoError(t, err)
 	return &tr
 }
+
+func TestInstance_TransactionPaymentCallApi_QueryCallInfo(t *testing.T) {
+	t.Parallel()
+	ins := NewTestInstance(t, runtime.WESTEND_RUNTIME_v0929)
+	tests := []struct {
+		callHex    string
+		errMessage string
+		expect     *types.RuntimeDispatchInfo
+	}{
+		{
+			// call generated by using palkadot.js/api v9.5.1: api.tx.system.remark("Ed")
+			// and removing first byte (encoding) and second byte (unknown)
+			callHex: "0x0001084564",
+			expect: &types.RuntimeDispatchInfo{
+				Weight: 0,
+				Class:  0,
+				PartialFee: &scale.Uint128{
+					Upper: 0,
+					Lower: uint64(1500000000),
+				},
+			},
+		},
+		{
+			// call removing encoding (first byte), polkadot.js/api v9.5.1: api.tx.system.remark("Ed")
+			// polkadot.js/api returns error:  RPC-CORE: call(method: Text, data: Bytes, at?: BlockHash):
+			// Bytes:: -32000: Client error: Execution failed: Execution aborted due to trap: wasm trap: wasm
+			//`unreachable` instruction executed
+			callHex: "0x040001084564",
+			errMessage: "running runtime function: " +
+				"Failed to call the `TransactionPaymentCallApi_query_call_info` exported function.",
+		},
+		{
+			// call without removing any bytes, polkadot.js/api v9.5.1: api.tx.system.remark("Ed test")
+			// polkadot.js/api returns error: Error: createType(Call):: findMetaCall: Unable to find Call with index
+			// [44, 4]/[44,4]
+			callHex: "0x2c0400011c45642074657374",
+			errMessage: "running runtime function: " +
+				"Failed to call the `TransactionPaymentCallApi_query_call_info` exported function.",
+		},
+	}
+
+	for _, test := range tests {
+		var err error
+		var callBytes []byte
+
+		callBytes, err = common.HexToBytes(test.callHex)
+		require.NoError(t, err)
+
+		info, err := ins.QueryCallInfo(callBytes)
+
+		if test.errMessage != "" {
+			assert.EqualError(t, err, test.errMessage)
+			continue
+		}
+
+		require.NoError(t, err)
+		require.NotNil(t, info)
+		require.Equal(t, test.expect, info)
+	}
+}
+
+func TestInstance_TransactionPaymentCallApi_QueryCallFeeDetails(t *testing.T) {
+	t.Parallel()
+	ins := NewTestInstance(t, runtime.WESTEND_RUNTIME_v0929)
+	tests := []struct {
+		callHex    string
+		errMessage string
+		expect     *types.FeeDetails
+	}{
+		{
+			// call generated by using palkadot.js/api v9.5.1: api.tx.system.remark("Ed")
+			// and removing first byte (encoding) and second byte (unknown)
```

Comment:
meaning we dont know what the second byte is?

Reply:
Correct, the second byte is 0x04, but I don't know why it's there are what it represents.

Reply:
I think it might represent this magic number https://github.com/paritytech/substrate/blob/polkadot-v0.9.30/primitives/runtime/src/generic/unchecked_extrinsic.rs#L34-L39